### PR TITLE
image preview bubble for data list view

### PR DESCRIPTION
**Pull request by tarafanlin**
- show the image preview when hover over the image name
- conditioned to only work for image files

to test this go to: http://localhost:1337/_?scene=data

Screen Shot 2020-10-20 at 2 22 25 AM

### CODE REVIEW

File: components/core/FilePreviewBubble.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import * as React from "react";
+import * as Constants from "~/common/constants";
+
+import { css, keyframes } from "@emotion/react";
+import { useState } from "react";
+import SlateMediaObjectPreview from "~/components/core/SlateMediaObjectPreview";
```
Comment (jimmylee) [inline components/core/FilePreviewBubble.js:6]: Nit: new line, fix this after we commit it.

File: components/core/FilePreviewBubble.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    opacity: 1;
+  }
+`;
+
+const STYLES_FILE_PREVIEW_BUBBLE = css`
+  z-index: ${Constants.zindex.tooltip};
+  padding: 8px;
+  border-radius: 4px;
+  background-color: ${Constants.system.black};
+  animation: ${fadein} 200ms ease-out 1;
```
Comment (jimmylee) [inline components/core/FilePreviewBubble.js:23]: nit: `fadeIn` and also update `const fadein` to `const fadeIn` above

File: components/core/FilePreviewBubble.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    <React.Fragment>
+      <div
+        css={STYLES_BUBBLE_ANCHOR}
+        onMouseEnter={() => setHover(true)}
+        onMouseLeave={() => setHover(false)}
+      >
+        {props.children}
+      </div>
+      {type && (
+        <div css={STYLES_FILE_PREVIEW_BUBBLE}>
```
Comment (jimmylee) [inline components/core/FilePreviewBubble.js:56]: Might want to leave a comment that the parent of this element needs to be `position: relative` or this bubble will fly off its position. or change the name of this component to `RelativeChildFilePreviewBubble`

File: components/core/FilePreviewBubble.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  padding: 8px;
+  border-radius: 4px;
+  background-color: ${Constants.system.black};
+  animation: ${fadein} 200ms ease-out 1;
+  width: 200px;
+  position: absolute;
+`;
+
+const STYLES_FILE_PREVIEW = css`
+  width: 100%;
```
Comment (jimmylee) [inline components/core/FilePreviewBubble.js:29]: `display: block`

File: components/core/FilePreviewBubble.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  height: 100%;
+`;
+
+const STYLES_BUBBLE_ANCHOR = css`
+  box-sizing: border-box;
```
Comment (jimmylee) [inline components/core/FilePreviewBubble.js:34]: `border-box` should be a natural style, you don't need to set it.

**Review by jimmylee — CHANGES_REQUESTED** [review]

**Comment by jimmylee** [comment]
Looking good! Just some details you should look at.

File: components/core/FilePreviewBubble.js (hunk 4 above)
Comment (tarafanlin) [reply components/core/FilePreviewBubble.js:29]: what's the consequence for not writing `display: block` ?

Comment (jimmylee) [reply components/core/FilePreviewBubble.js:28]: block is width naturally 100%, height 100% doesn't do anything, and then the other two rules are not important anymore.

**Comment by jimmylee** [comment]
LGTM

**Pull request merged**